pkg/network/node/subnets.go

```diff
@@ -41,7 +41,12 @@ func (node *OsdnNode) getLocalSubnet() (string, error) {
 		var err error
 		subnet, err = node.networkClient.Network().HostSubnets().Get(node.hostName, metav1.GetOptions{})
 		if err == nil {
-			return true, nil
+			if subnet.HostIP == node.localIP {
+				return true, nil
+			} else {
+				glog.Warningf("Waiting for latest local subnet for node: %s ...", node.hostName)
```

Comment:
Maybe make this more explicit so it's clearer what's happening if something goes wrong?

"HostIP for HostSubnet %q does not match node IP... waiting for master to update the HostSubnet" ?

Although could we get away with just doing "subnet.HostIP = node.localIP" here (and still logging a warning just in case). That has the same expected eventual result; if a HostSubnet already exists then the master isn't going to change the .Subnet value anyway, just the .HostIP...

Reply:
Yes, I will make the warning more clear. 
I feel we should not do "subnet.HostIP = node.localIP" here. Things like this can mask original issues. For example, if master has a bug and it didn't update the hostIP for the subnet, this node will not have issues but other nodes in the cluster will have incorrect/stale ovs rules for this referring node. Waiting for the latest subnet for node during getLocalSubnet() can uncover this bug quickly.